independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/RequestMapper.java

```diff
@@ -43,24 +43,46 @@ public void accept(String stem, ArrayList<RequestPath<T>> list) {
     }
 
     public RequestMatch<T> map(String path) {
-        var result = mapFromPathMatcher(path, requestPaths.match(path));
-        if (result != null) {
+        var result = mapFromPathMatcher(path, requestPaths.match(path), true);
+        if (!result.isEmpty()) {
+            return result.get(0);
+        }
+
+        // the following code is meant to handle cases like https://github.com/quarkusio/quarkus/issues/30667
+        result = mapFromPathMatcher(path, requestPaths.defaultMatch(path), true);
+        if (!result.isEmpty()) {
+            return result.get(0);
+        }
+        return null;
+    }
+
+    /**
+     * Retrieve all UriTemplate matches for a given path, ordered by how much of the path is remaining
+     *
+     * @param path path to search UriTemplate for
+     * @return list, never null, may be empty
+     */
+    public List<RequestMatch<T>> allMatches(String path) {
+        var result = mapFromPathMatcher(path, requestPaths.match(path), false);
+        if (!result.isEmpty()) {
             return result;
         }
 
         // the following code is meant to handle cases like https://github.com/quarkusio/quarkus/issues/30667
-        return mapFromPathMatcher(path, requestPaths.defaultMatch(path));
+        return mapFromPathMatcher(path, requestPaths.defaultMatch(path), false);
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    private RequestMatch<T> mapFromPathMatcher(String path, PathMatcher.PathMatch<ArrayList<RequestPath<T>>> initialMatch) {
+    private List<RequestMatch<T>> mapFromPathMatcher(String path,
```

Comment:
Please have a version of this method which doesn't return a collection, but a single value - since is what is been used in some hot path 🙏

Reply:
done